### PR TITLE
first pass at a doc about upgrading dependencies

### PR DESCRIPTION
**Pull request by shiftkey**
## Overview

[**Rendered Markdown**](https://github.com/desktop/desktop/blob/upgrade-dependencies-doc/docs/contributing/upgrading-dependencies.md)

This document came out of a discussion I had with @say25 a couple of weeks ago about how the project manages dependencies.

## Description

For the moment I'm just focused on outlining how we upgrade dependencies. We did talk about automation services of these like [Dependabot](https://dependabot.com/) or [Greenkeeper](https://greenkeeper.io/) but I think the process should be clarified first.

## Release notes

Notes: no-notes

### CODE REVIEW

File: docs/contributing/upgrading-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+contribution is opened by the community.
+
+### Application dependencies
+
+This group of dependencies represent the rest of the listed dependencies in
+`app/pacakge.json` - there are lots of small dependencies which are approachable
+for external contributions.
+
+If you are interested in upgrading one of these, take a look at where it is
+used in the application and confirm the affected functionality works as
```
Comment (say25) [inline docs/contributing/upgrading-dependencies.md]:
```suggestion
used in the application and confirm that the affected functionality works as
```

File: docs/contributing/upgrading-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+expected before opening a pull request.
```
Comment (say25) [inline docs/contributing/upgrading-dependencies.md]: Is it work mentioning about the "test" releases here i.e. `1.4.1-test1`?

File: docs/contributing/upgrading-dependencies.md (hunk 1 above)
Comment (say25) [reply docs/contributing/upgrading-dependencies.md]: minor thing not really a big deal. I think it reads a little better ¯\_(ツ)_/¯

File: docs/contributing/upgrading-dependencies.md (hunk 2 above)
Comment (shiftkey) [reply docs/contributing/upgrading-dependencies.md]: I don't think so. I consider those test releases special snowflake builds that we might use to verify a specific fix (it's often easier to link an installer to someone than to get them to work with the source) but beyond that I consider them unofficial.

**Pull request merged**